### ISSUE TITLE
Use Session scoped fixture instead of global

### ISSUE THREAD

**Issue by danpalmer** [csernazs/pytest-httpserver#32]
I may be missing a particular design reason that means this is not possible, however I think it could be beneficial to make the `httpserver` fixture a session-scoped fixture rather than use a global variable to store it.

By making it session-scoped it will only be created once as needed, which I believe is the intention of the global variable. This also means that other session-scoped fixtures can use it, where currently they are unable to.

**Comment by csernazs** [comment]
Hmm, probably I missed this feature of the pytest when I wrote the library - the global variable was introduced when I realized that starting/stopping the server takes 0.1 sec time at minimum, so it is better to have the server started.

I think you are completely right, this could be a session-scoped fixture (and this should have been, my mistake). My only concern is backward compatibility. Can we do this by replacing the scope of the fixture or should we introduce a new fixture with session scope? What do you think about it?

I can make the PR for this tomorrow if that's ok for you.

**Comment by csernazs** [comment]
I looked at the code today and I found that having session scoped fixture is a no-go in this case.
The problem is that between test runs the server's state needs to be cleared. The server stores all the handlers the user registered, and those should be handled separately between the tests - there should be no cross talk.

I do want to understand why it would be still important for you, maybe we can find a solution.

There's currently a `pytest_sessionfinish` function provided in the code which gets called when the session ends, in order to stop the server and do a final cleanup, this could be moved to a session-scoped fixture.
But in this case the httpserver fixture would be still function scoped.
